node/core/candidate-validation/src/lib.rs

```diff
@@ -318,25 +359,52 @@ where
 			},
 		};
 
-	let executor_params =
-		if let Ok(executor_params) = executor_params_at_relay_parent(relay_parent, sender).await {
+	let executor_params = match request_pending_executor_params(sender, relay_parent).await {
+		Ok(executor_params) => {
 			gum::debug!(
 				target: LOG_TARGET,
 				?relay_parent,
 				?validation_code_hash,
-				"precheck: acquired executor params for the session: {:?}",
+				"precheck: acquired pending executor params: {:?}",
 				executor_params,
 			);
 			executor_params
-		} else {
+		},
+		Err(RuntimeRequestFailed::NotSupported) => {
+			// Old runtime that doesn't support retrieving pending configuration. Use the
+			// current configuration as a backward compatibility measure.
+			// TODO: Remove this after all the networks are upgraded to runtimes supporting
+			//       pending_executor_params()
+			if let Ok(executor_params) = executor_params_at_relay_parent(relay_parent, sender).await
+			{
+				gum::debug!(
+					target: LOG_TARGET,
+					?relay_parent,
+					?validation_code_hash,
+					"precheck: acquired current executor params: {:?}",
+					executor_params,
+				);
+				executor_params
+			} else {
```

Comment:
What about not supported?

Reply:
It's [handled](https://github.com/paritytech/polkadot/blob/689623c97071d9fac951a459f0b5100b34697a62/node/subsystem-util/src/lib.rs#L247) inside the `executor_params_at_relay_parent()` itself.